prog/weavewait/main.go

```diff
@@ -39,3 +61,43 @@ func checkErr(err error) {
 		os.Exit(1)
 	}
 }
+
+// Prepend the addresses to /etc/hosts, removing the docker address
+func updateHosts(iface *net.Interface) {
```

Comment:
I think it might be best to try a different approach with this function.  This current version is too hard to follow and maintain IMO.

I would construct, in memory, what you want the file to be, maybe as a [][2]string.  Read the current contents, remove the entries you don't want, add the entries you do, sort(?) then write out to disk.